pool.go

```diff
@@ -45,6 +46,30 @@ type ClientConn struct {
 	timeUsed      time.Time
 	timeInitiated time.Time
 	unhealthy     bool
+	concurrency   *ConcurrencyCounter
+}
+
+type ConcurrencyCounter struct {
+	concurrency int `default:0`
+	mu          sync.RWMutex
+}
+
+func (c *ConcurrencyCounter) Get() int {
+	return c.concurrency
```

Comment:
Hmm, seems like we might not be using a rwlock correctly. It seems like we're using a read-lock to protect writes, and not using the write lock at all.

Shouldn't we wrap the get in read-lock, and inc/dec in write-lock?